client/src/components/AddMates/AddMatesPage.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Button,
+  Col,
+  DatePicker,
+  Divider,
+  Form,
+  Input,
+  Layout,
+  Row,
+  Space,
+  Typography,
+} from "antd";
+import React, { FC, useState } from "react";
+import { Prompt } from "react-router";
+
+import { AddMateFormType, AddMateType } from "../types";
+import AddMateCard from "./AddMateCard";
+
+const { Title } = Typography;
+const { Footer } = Layout;
+
+const AddMatesPage: FC = () => {
+  const [form] = Form.useForm<AddMateFormType>();
+  const [mates, setMates] = useState<AddMateType[]>([]);
+
+  const onAddMate = async ({ name, lastSeen }: AddMateFormType) => {
+    setMates((prevArray) => [
+      ...prevArray,
+      { name, lastSynced: lastSeen.toDate(), ts: new Date().getTime() },
+    ]);
+    form.resetFields();
+  };
+
+  const removeMate = (mate: AddMateType) => {
+    setMates((mateArr) =>
+      mateArr.filter(
+        (m8) =>
+          !(
+            m8.name === mate.name &&
+            m8.lastSynced === mate.lastSynced &&
+            m8.ts === mate.ts
+          )
+      )
+    );
+  };
+
+  const handleAddMates = () => {
+    // TODO send to backend here
+  };
+
+  const cards = mates.map((mate) => (
+    <Col span={6} key={mate.ts}>
+      <AddMateCard mate={mate} removeMate={removeMate} />
+    </Col>
+  ));
+
+  const pageUnsaved = mates.length > 0;
+
+  return (
+    <>
+      <Prompt
+        when={pageUnsaved}
+        message="Leaving will erase all your unsaved info. Are you sure?"
+      />
+      <Layout style={{ minHeight: "100vh" }}>
+        <Row style={{ backgroundColor: "#F0F2F5", padding: "30px" }}>
+          <Col>
+            <Title>Populating M8s</Title>
+          </Col>
+        </Row>
+        <Row style={{ padding: "30px" }}>
+          <Col span={6}>
+            <Title level={2}>Manual Entry</Title>
+          </Col>
+
+          <Col>
+            <Form layout="inline" form={form} onFinish={onAddMate}>
+              <Form.Item label="Name" name="name" rules={[{ required: true }]}>
+                <Input autoFocus />
+              </Form.Item>
+              <Form.Item
+                label="Last Seen"
+                name="lastSeen"
+                rules={[{ required: true }]}
```

Comment:
might be kind of annoying to fill every time, esp since i might not remember; what do you think about assigning it some default value if you dont have anything set when you press 'add'?

Reply:
defaults to today for now.